core/editor/src/input/mouse.rs

```diff
@@ -0,0 +1,48 @@
+use bitflags::bitflags;
+
+// origin is top left
+#[derive(Debug, Copy, Clone, Default, Eq, PartialEq)]
+pub struct ViewportPosition {
+	pub x: u32,
+	pub y: u32,
+}
+
+impl ViewportPosition {
+	pub fn distance(&self, other: &Self) -> f64 {
+		let x_diff = other.x as i32 - self.x as i32;
+		let y_diff = other.y as i32 - self.y as i32;
+		f64::sqrt((x_diff * x_diff + y_diff * y_diff) as f64)
```

Comment:
This needs `f64::hypot`

Reply:
in that case you have to do two conversions to f64

Reply:
You have to do two conversions to f64 but I am not qualified enough to assess the impact that has on each plattform
``` 
        mov     eax, edi
        cvtsi2sd        xmm0, rax
        movapd  xmm1, xmm0
        jmp     qword ptr [rip + hypot@GOTPCREL]
  ```
  vs
  ```
  example::square:
        imul    edi, edi
        imul    esi, esi
        add     esi, edi
        cvtsi2sd        xmm0, rsi
        sqrtsd  xmm0, xmm0
        ret
      ```

Reply:
I think hpot might actually be slower: https://github.com/rust-lang/rust/issues/41908
Given we can expect to have very tame inputs I don't actually think hpot would provide that much benefit. But as always micro optimisations without benchmarking are kinda pointless